### PR TITLE
[Chore] Upgrade to MeshKit v0.5.16

### PR DESCRIPTION
**Pull request by leecalcote**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,152 +5,22 @@ go 1.17
...
+	k8s.io/apimachinery => k8s.io/apimachinery v0.22.8
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.22.8
+	k8s.io/client-go => k8s.io/client-go v0.22.8
+	k8s.io/kubectl => k8s.io/kubectl v0.22.8
 )
 
 require (
 	github.com/go-logr/logr v0.4.0
-	github.com/layer5io/meshkit v0.5.2
+	github.com/layer5io/meshkit v0.5.15
```
Comment (jbyers19) [inline go.mod:19]: Meshkit v0.5.16 was just released.

**Comment by leecalcote** [comment]
Upgrade to MeshKit v0.5.16: failing.

**Comment by jbyers19** [comment]
> Upgrade to MeshKit v0.5.16: failing.

@leecalcote, looks like there's a discrepancy between `go.mod` and `go.sum`. Can you try running `go mod tidy` and see if that fixes it?

**Comment by Aisuko** [comment]
> Upgrade to MeshKit v0.5.16: failing.

Hi @leecalcote Do you mean the CI error?

**Comment by Aisuko** [comment]
I guess the error of CI's log was trigger by the specific mesh-kit. If we upgrade to the v0.5.16. I believe it will be ok.

**Review by Aisuko — COMMENTED** [review]
Let's fix the issue of the CI's process.

**Comment by leecalcote** [comment]
@jbyers19 good ask. Yes, I have run go mod tidy, which is also included in the `make` command.

**Comment by Aisuko** [comment]
> @jbyers19 good ask. Yes, I have run go mod tidy, which is also included in the `make` command.

Look like we almost finished this.

**Comment by jbyers19** [comment]
@leecalcote , @Aisuko: looks like there is a type issue with the logger changes I introduced in meshkit `v0.5.14`. I will work on getting all the dependencies updated.

File: go.mod [hunk 2]
```diff
@@ -5,152 +5,22 @@ go 1.17
 replace (
 	github.com/kudobuilder/kuttl => github.com/layer5io/kuttl v0.4.1-0.20200806180306-b7e46afd657f
 	vbom.ml/util => github.com/fvbommel/util v0.0.0-20180919145318-efcd4e0f9787
-// golang.org/x/sys => golang.org/x/sys v0.0.0-20200826173525-f9321e4c35a6
+//	golang.org/x/sys => golang.org/x/sys v0.0.0-20220319134239-a9b59b0215f8
```
Comment (jbyers19) [inline go.mod:8]: Is there a need to keep this commented out? Would be nice to delete it if it's not being used.

File: go.mod [hunk 3]
```diff
@@ -5,152 +5,22 @@ go 1.17
 replace (
 	github.com/kudobuilder/kuttl => github.com/layer5io/kuttl v0.4.1-0.20200806180306-b7e46afd657f
 	vbom.ml/util => github.com/fvbommel/util v0.0.0-20180919145318-efcd4e0f9787
```
Comment (jbyers19) [inline go.mod:7]: Doesn't look like this is being used any more and can be removed.

File: go.mod (hunk 2 above)
Comment (leecalcote) [reply go.mod:8]: No good reason, no. If memory serves, I think people were just fearful of losing it in the event a revert was needed.

File: go.mod [hunk 4]
```diff
@@ -5,152 +5,22 @@ go 1.17
...
-// golang.org/x/sys => golang.org/x/sys v0.0.0-20200826173525-f9321e4c35a6
+//	golang.org/x/sys => golang.org/x/sys v0.0.0-20220319134239-a9b59b0215f8
+	github.com/Sirupsen/logrus => github.com/sirupsen/logrus v1.8.1
+	k8s.io/api => k8s.io/api v0.22.8
+	k8s.io/apimachinery => k8s.io/apimachinery v0.22.8
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.22.8
+	k8s.io/client-go => k8s.io/client-go v0.22.8
+	k8s.io/kubectl => k8s.io/kubectl v0.22.8
```
Comment (jbyers19) [inline go.mod:14]: `k8s.io/kubectl => k8s.io/kubectl v0.22.8` can be removed, but these two need to be added:
```
	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.22.8
	k8s.io/component-base => k8s.io/component-base v0.22.8
```

File: go.mod [hunk 5]
```diff
@@ -5,152 +5,22 @@ go 1.17
...
 require (
 	github.com/go-logr/logr v0.4.0
-	github.com/layer5io/meshkit v0.5.2
+	github.com/layer5io/meshkit v0.5.16
 	github.com/onsi/ginkgo v1.16.5
 	github.com/onsi/gomega v1.18.1
 	k8s.io/api v0.23.0-alpha.1
 	k8s.io/apimachinery v0.23.0-alpha.1
 	k8s.io/client-go v0.23.0-alpha.1
-	sigs.k8s.io/controller-runtime v0.10.3
```
Comment (jbyers19) [inline go.mod:19]: This needs to be: `sigs.k8s.io/controller-runtime v0.11.2`

File: go.mod [hunk 6]
```diff
@@ -5,152 +5,22 @@ go 1.17
...
+	github.com/Sirupsen/logrus => github.com/sirupsen/logrus v1.8.1
+	k8s.io/api => k8s.io/api v0.22.8
+	k8s.io/apimachinery => k8s.io/apimachinery v0.22.8
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.22.8
+	k8s.io/client-go => k8s.io/client-go v0.22.8
+	k8s.io/kubectl => k8s.io/kubectl v0.22.8
 )
 
 require (
 	github.com/go-logr/logr v0.4.0
```
Comment (jbyers19) [inline go.mod:18]: This needs to be: `github.com/go-logr/logr v1.2.3`

Comment (leecalcote) [reply go.mod:18]: Thank you for this, @jbyers19

**Comment by leecalcote** [comment]
This has been updated to MeshKit v0.5.17. I _think_ all comments have been addressed. Please note the changes to the custom resources and manifest.

File: go.mod (hunk 4 above)
Comment (leecalcote) [reply go.mod]: This is most helpful. Thanks for this.

**Review by jbyers19 — APPROVED** [review]
LGTM and all the checks are passing now. 🎉

**Comment by leecalcote** [comment]
yay! 😄

**Pull request merged**